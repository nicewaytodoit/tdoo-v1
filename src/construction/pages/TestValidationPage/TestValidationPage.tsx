import React from 'react';

export interface IValidationProps { id: string, name?: any; value?: any; }

class TestValidationPage extends React.Component<IValidationProps, any> {
    state = {
        name: '',
        age: 12,
        email: null,
        isNameValid: true,
    }

    NameChange = (e) => {
        this.setState({ name: e.target.value });
    }

    Validate = (val, prop) => {
        this.setState({ [`isNameValid`]: !val });
    }

    render() {
        return (
            <div>
                <h1>Validation Test</h1>
                <input value={this.state.name} onChange={this.NameChange} onBlur={(e) => this.Validate(e.target.value, 'name')} style={{ borderColor: this.state.isNameValid ? 'red' : 'green' }} />
                { !this.state.isNameValid ?<label> Empty </label> : '' }
            </div>
        );
    }

}

export default TestValidationPage;