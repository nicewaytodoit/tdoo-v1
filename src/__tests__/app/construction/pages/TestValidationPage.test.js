import React from 'react';
import TestValidationPage from '../../../../construction/pages/TestValidationPage/TestValidationPage';
import { shallow } from 'enzyme';


describe('Page: TestValidationPage', () => {

    const control = shallow(<TestValidationPage />);

    it('should match snapshot', () => {
        expect(control).toMatchSnapshot();
    });

    it('should have h1 element', () => {
        expect(control.find('h1')).toBeTruthy();
    });
});