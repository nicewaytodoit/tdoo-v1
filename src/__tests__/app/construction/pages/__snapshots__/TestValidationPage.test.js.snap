// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: TestValidationPage should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TestValidationPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Validation Test
        </h1>,
        <input
          onBlur={[Function]}
          onChange={[Function]}
          style={
            Object {
              "borderColor": "red",
            }
          }
          value=""
        />,
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Validation Test",
        },
        "ref": null,
        "rendered": "Validation Test",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onBlur": [Function],
          "onChange": [Function],
          "style": Object {
            "borderColor": "red",
          },
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Validation Test
          </h1>,
          <input
            onBlur={[Function]}
            onChange={[Function]}
            style={
              Object {
                "borderColor": "red",
              }
            }
            value=""
          />,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Validation Test",
          },
          "ref": null,
          "rendered": "Validation Test",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onBlur": [Function],
            "onChange": [Function],
            "style": Object {
              "borderColor": "red",
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
